src/main/java/org/embulk/input/marketo/delegate/ProgramInputPlugin.java

```diff
@@ -179,14 +189,18 @@ public ConfigDiff buildConfigDiff(PluginTask task, Schema schema, int taskCount,
         ConfigDiff configDiff = super.buildConfigDiff(task, schema, taskCount, taskReports);
         // set next next earliestUpdatedAt, latestUpdatedAt
         if (task.getQueryBy().isPresent() && task.getQueryBy().get() == QueryBy.DATE_RANGE && task.getIncremental()) {
-            DateTime earliest = new DateTime(task.getEarliestUpdatedAt().get());
-            DateTime latest = new DateTime(task.getLatestUpdatedAt().get());
-
-            Duration d = new Duration(earliest, latest);
-            DateTime nextEarliestUpdatedAt = latest.plusSeconds(1);
-
-            configDiff.set("earliest_updated_at", nextEarliestUpdatedAt.toString(DATE_FORMATTER));
-            configDiff.set("report_duration", task.getReportDuration().or(d.getMillis()));
+            OffsetDateTime earliest = task.getEarliestUpdatedAt().isPresent() ?
+                    task.getEarliestUpdatedAt().get().toInstant().atOffset(ZoneOffset.UTC) :
+                    OffsetDateTime.now(ZoneOffset.UTC);
+            OffsetDateTime latest = task.getLatestUpdatedAt().isPresent() ?
+                    task.getLatestUpdatedAt().get().toInstant().atOffset(ZoneOffset.UTC) :
+                    OffsetDateTime.now(ZoneOffset.UTC);
```

Comment:
ditto, redundant check for presentation of the two keys already checked in validation method